core/metacling/src/TCling.cxx

```diff
@@ -5891,11 +5869,98 @@ Int_t TCling::AutoParse(const char *cls)
    return nHheadersParsed > 0 ? 1 : 0;
 }
 
+void* TCling::LazyFunctionCreatorAutoloadForModule(const std::string& mangled_name) {
+   using namespace llvm::object;
+   using namespace llvm::sys::path;
+   using namespace llvm::sys::fs;
+
+   static constexpr std::hash<std::string> hash_fn;
+   static std::unordered_map<size_t, uint8_t> libtable;
+   static std::vector<std::string> int_and_libtable;
+
+   // Initialize the table
+   if (fFirstRun) {
+      clang::Preprocessor &PP = fInterpreter->getCI()->getPreprocessor();
+      const HeaderSearchOptions &HSOpts = PP.getHeaderSearchInfo().getHeaderSearchOpts();
+      auto ModulePaths(HSOpts.PrebuiltModulePaths);
+      std::sort(ModulePaths.begin(), ModulePaths.end());
+      auto last = std::unique(ModulePaths.begin(), ModulePaths.end());
+      ModulePaths.erase(last, ModulePaths.end());
+
+      cling::DynamicLibraryManager* dyLibManager = fInterpreter->getDynamicLibraryManager();
+
+      int cur_lib = 0;
+      for (auto Path : ModulePaths) {
+         StringRef DirPath(Path);
+         if (DirPath.empty() || !is_directory(DirPath) || Path == ".")
+            continue;
+
+         // Check if this is a build directory (which should contain pcm)
+         std::error_code EC;
+         bool is_build = false;
+         for (llvm::sys::fs::directory_iterator DirIt(DirPath, EC), DirEnd;
+               DirIt != DirEnd && !EC; DirIt.increment(EC)) {
+            std::string FileName(DirIt->path());
+            if (extension(FileName) == ".pcm") {
+               is_build = true;
+               break;
+            }
+         }
+
+         if (!is_build)
+            continue;
+
+         // Iterate over the directory
+         for (llvm::sys::fs::directory_iterator DirIt(DirPath, EC), DirEnd;
+               DirIt != DirEnd && !EC; DirIt.increment(EC)) {
+            std::string LibName(DirIt->path());
+            if (llvm::sys::fs::is_directory(LibName) || extension(LibName) != ".so")
+               continue;
+
+            // TCling::IsLoaded is incredibly slow!
+            if (dyLibManager->isLibraryLoaded(LibName.c_str()))
+               continue;
+
+            int_and_libtable.push_back(LibName);
+            auto SoFile = ObjectFile::createObjectFile(LibName);
+            if (SoFile) {
+               auto RealSoFile = SoFile.get().getBinary();
+               auto Symbols = RealSoFile->symbols();
+               for (auto S : Symbols) {
+                  // DO NOT load to table if symbol was weak or undefined
+                  if (S.getFlags() == SymbolRef::SF_Weak || S.getFlags() == SymbolRef::SF_Undefined) continue;
+
+                  std::string SymName = std::string(S.getName().get());
+                  // Put into map
+                  libtable.insert(std::make_pair(hash_fn(SymName), cur_lib));
```

Comment:
Do we know how much memory does this data structure take when it is fully initialized?